shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-provider/shardingsphere-db-discovery-mgr/src/main/java/org/apache/shardingsphere/dbdiscovery/mgr/MGRDatabaseDiscoveryType.java

```diff
@@ -114,6 +115,31 @@ private void checkIsSinglePrimaryMode(final Statement statement) throws SQLExcep
         }
     }
     
+    private void checkDataSourceInReplicationGroup(final Statement statement, final Map<String, DataSource> dataSourceMap) throws SQLException {
+        Collection<String> memberDataSourceURLs = new LinkedList<>();
+        try (ResultSet resultSet = statement.executeQuery(MEMBER_LIST)) {
+            while (resultSet.next()) {
+                memberDataSourceURLs.add(String.format("%s:%s", resultSet.getString("MEMBER_HOST"), resultSet.getString("MEMBER_PORT")));
+            }
+        }
+        for (Entry<String, DataSource> entry : dataSourceMap.entrySet()) {
+            checkDataSourceExistGroupMember(entry.getKey(), entry.getValue(), memberDataSourceURLs);
+        }
+    }
+    
+    private void checkDataSourceExistGroupMember(final String datasourceName, final DataSource dataSource, final Collection<String> memberDataSourceURLs) throws SQLException {
+        boolean exist = false;
```

Comment:
`exist` is a verb, how about rename it as `isExisted`?

Reply:
Done.